tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -478,3 +552,71 @@ def _make_tf_features(self, input_feat):
                                   [-1, self._feature_size])
       freq_inputs.append(cur_input)
     return freq_inputs
+
+
+class LSTMCellBlock(rnn_cell.RNNCell):
+  """Basic LSTM recurrent network cell.
+
+  The implementation is based on: http://arxiv.org/abs/1409.2329.
+
+  We add forget_bias (default: 1) to the biases of the forget gate in order to
+  reduce the scale of forgetting in the beginning of the training.
+
+  It does not allow cell clipping, a projection layer, and does not
+  use peep-hole connections: it is the basic baseline.
+
+  Unlike BasicLSTMCell, this is a monolithic op and should be much faster. The
+  weight and bias matrixes should be compatible as long as the variabel scope
+  matches.
+  """
+
+  def __init__(self, num_units, forget_bias=1.0, input_size=None, bprop_dx=True,
+               parallel_dw=True):
+    """Initialize the basic LSTM cell.
+
+    Args:
+      num_units: int, The number of units in the LSTM cell.
+      forget_bias: float, The bias added to forget gates (see above).
+    """
+    if input_size is not None:
+      logging.warn("%s: The input_size parameter is deprecated." % self)
+    self._num_units = num_units
+    self._forget_bias = forget_bias
+    self._bprop_dx = bprop_dx
+    self._parallel_dw = parallel_dw
+
+  @property
+  def state_size(self):
+    return self._num_units * 2
+
+  @property
+  def output_size(self):
+    return self._num_units
+
+  def __call__(self, x, states_prev, scope=None):
+    """Long short-term memory cell (LSTM)."""
+    with vs.variable_scope(scope or type(self).__name__):
+      w = vs.get_variable("W", [x.get_shape()[1] + self._num_units,
```

Comment:
x_shape = x.get_shape().with_rank(2)
check that x_shape[1] is not None or raise a ValueError.  same with w below.